### PR TITLE
create vertical icons panel react lib

### PR DESCRIPTION
**Pull request by joeizang**
fix https://github.com/ethereum/remix-project/issues/1333
fix https://github.com/ethereum/remix-project/issues/133

### CODE REVIEW

**Comment by ryestew** [comment]
I'm still having troubles getting the scroll to work properly.  Also - and this may just have to do wtih IPFS - when I try to load vyper - I see the icon in the icon panel but it does move to the active list in the plugin manager.

**Comment by joeizang** [comment]
> I'm still having troubles getting the scroll to work properly. Also - and this may just have to do wtih IPFS - when I try to load vyper - I see the icon in the icon panel but it does move to the active list in the plugin manager.

I experienced both also but only on IPFS. I concluded that its an ipfs thing with plugins not working properly. This was an issue @bunsenstraat mentioned before and alluded to the fact that our plugins need to be updated somehow.

**Comment by LianaHus** [comment]
sorry but I found a problem in the PR. The 'loading' badge on solidity icon is shifted. try to load a compiler to see the problem
@joeizang

**Comment by Aniket-Engg** [comment]
image

I still see this on IPFS link.

**Comment by LianaHus** [comment]
also I think we should discuss the position of plugin manager. doesn't look right in the middle of icons. I would suggest to

- either put it after the FE icon
- have a fine line after it to show that all native plugins are above it
- always show it at the end, before the settings icon(as we have now) even when we have the view with separators.

let's discuss this after the standup today

**Review by Aniket-Engg — COMMENTED** [review]
There are multiple logs in browser with text `new status for solidity is :` and `from reducer of solidity :`

**Comment by joeizang** [comment]
> There are multiple logs in browser with text `new status for solidity is :` and `from reducer of solidity :`

they have since been removed.

**Comment by joeizang** [comment]
> also I think we should discuss the position of plugin manager. doesn't look right in the middle of icons. I would suggest to
> 
> * either put it after the FE icon
> * have a fine line after it to show that all native plugins are above it
> * always show it at the end, before the settings icon(as we have now) even when we have the view with separators.
> 
> Screenshot from 2021-11-09 10-58-20
> 
> let's discuss this after the standup today

This is done.

**Comment by joeizang** [comment]
> sorry but I found a problem in the PR. The 'loading' badge on solidity icon is shifted. try to load a compiler to see the problem @joeizang

This also has been fixed.

File: apps/remix-ide-e2e/src/helpers/init.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ require('dotenv').config()
 export default function (browser: NightwatchBrowser, callback: VoidFunction, url?: string, preloadPlugins = true): void {
   browser
     .url(url || 'http://127.0.0.1:8080')
-    .pause(5000)
+    .pause(10000)
```
Comment (ioedeveloper) [inline apps/remix-ide-e2e/src/helpers/init.ts:8]: `init.ts` is used in all test files. A 5sec increase in delay will increase circleci time significantly.

File: apps/remix-ide-e2e/src/tests/plugin_api.ts [hunk 2]
```diff
@@ -112,6 +112,7 @@ const clickAndCheckLog = async (browser: NightwatchBrowser, buttonText: string,
   if (methodResult && typeof methodResult !== 'string') { methodResult = JSON.stringify(methodResult) }
   if (eventResult && typeof eventResult !== 'string') { eventResult = JSON.stringify(eventResult) }
   if (buttonText) {
+    console.log('this is buttonText in this test', buttonText)
```
Comment (ioedeveloper) [inline apps/remix-ide-e2e/src/tests/plugin_api.ts:115]: Is this log needed? If not, please remove.

File: package.json [hunk 3]
```diff
@@ -155,6 +155,7 @@
     "chokidar": "^2.1.8",
     "color-support": "^1.1.3",
     "commander": "^2.20.3",
+    "core-js": "^3.6.5",
```
[the remark this answers is not in the document]
Comment (ioedeveloper) [reply package.json:161]: I still see `core-js` listed as a dependency.

File: workspace.json [hunk 4]
```diff
@@ -1035,72 +1101,36 @@
...
+    "@remix-ui-verticaliconspanel": {
+      "root": "libs/@remix-ui/verticaliconspanel",
+      "sourceRoot": "libs/@remix-ui/verticaliconspanel/src",
+      "projectType": "library",
+      "architect": {
+        "lint": {
+          "builder": "@nrwl/linter:eslint",
+          "options": {
+            "lintFilePatterns": [
+              "libs/@remix-ui/verticaliconspanel/**/*.{ts,tsx,js,jsx}"
```
[the remark this answers is not in the document]
Comment (ioedeveloper) [reply workspace.json:1121]: I still see 2 entries for `@remix-ui-verticaliconspanel` in workspace.json

**Review by ioedeveloper — CHANGES_REQUESTED** [review]

File: apps/remix-ide-e2e/src/tests/plugin_api.ts (hunk 2 above)
Comment (joeizang) [reply apps/remix-ide-e2e/src/tests/plugin_api.ts:115]: sorry left this while working here.

File: apps/remix-ide-e2e/src/helpers/init.ts (hunk 1 above)
Comment (joeizang) [reply apps/remix-ide-e2e/src/helpers/init.ts:8]: I have made this value 6000, I hope this is an acceptable value.

File: workspace.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (joeizang) [reply workspace.json:1121]: I have cleaned these up now.

File: libs/remix-ui/vertical-icons-panel/src/lib/components/RequiredSection.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  itemContextAction: (e: any, name: string, documentation: string) => Promise<void>
+  addActive: (name: string) => void
+  removeActive: () => void
+  scrollableRef: MutableRefObject<any>
+}
+
+function RequiredSection ({ verticalIconsPlugin, itemContextAction, addActive, removeActive, scrollableRef }: RequiredSectionProps) {
+  return (
+    <Fragment>
+      <FilePanel
```
Comment (ioedeveloper) [inline libs/remix-ui/vertical-icons-panel/src/lib/components/RequiredSection.tsx:23]: As an improvement, we can refactor the required section to reuse a single Icon component with changing props.

**Review by ioedeveloper — APPROVED** [review]

**Review by ryestew — APPROVED** [review]

**Pull request merged**